### ISSUE TITLE
Migrate functional-red-black-tree to js-sdsl

### ISSUE THREAD

**Issue by ZLY201** [ethereumjs/ethereumjs-monorepo#2283]
Hey! I'm the developer of Js-sdsl. Official website: https://js-sdsl.github.io/

Now, we published the version 4.1.4.

I see you are using functional-red-black-tree.

In [benchmark](https://js-sdsl.github.io/#/test/benchmark-result?id=rbtree), we have confirmed that Js-sdsl is several times faster than functional-red-black-tree.

I've submitted a PR to eslint and has been accepted.

See https://github.com/eslint/eslint/issues/16255 and https://github.com/eslint/eslint/pull/16267.

I am willing to submit a pull request for this change.

Looking forward to your reply! :D

**Comment by acolytec3** [comment]
Thanks for the note. We're definitely open to a PR though will want to see the benchmarks using our own code before merging. Note, this dependency is also used in our ` statemanager` package.

**Comment by ZLY201** [comment]
OK, I'll show you the benchmark result later.

**Comment by ZLY201** [comment]
Benchmarks here:

- perf/migrate-rbtree

https://github.com/ZLY201/ethereumjs-monorepo/actions/runs/3054050866/jobs/4925454999

```bash
> @ethereumjs/vm@6.0.0 benchmarks
> node --max-old-space-size=4096 ./benchmarks/run.js benchmarks mainnetBlocks:10

Benchmarking started...
Running 'mainnetBlocks':
Total number of blocks in data set: 15
Number of blocks to sample: 10
Block 9422905 x 1[8](https://github.com/ZLY201/ethereumjs-monorepo/actions/runs/3054050866/jobs/4925454999#step:6:9),375 ops/sec ±2.02% (82 runs sampled)
Block [9](https://github.com/ZLY201/ethereumjs-monorepo/actions/runs/3054050866/jobs/4925454999#step:6:10)422906 x 17,719 ops/sec ±4.70% (82 runs sampled)
Block 9422907 x 17,477 ops/sec ±4.75% (85 runs sampled)
Block 9422908 x 16,721 ops/sec ±9.67% (85 runs sampled)
Block 94229[10](https://github.com/ZLY201/ethereumjs-monorepo/actions/runs/3054050866/jobs/4925454999#step:6:11) x [17](https://github.com/ZLY201/ethereumjs-monorepo/actions/runs/3054050866/jobs/4925454999#step:6:18),523 ops/sec ±1.95% (80 runs sampled)
Benchmark run finished.
```

- master

https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/3049324777/jobs/4915279042

```bash
> @ethereumjs/vm@6.0.0 benchmarks
> node --max-old-space-size=4096 ./benchmarks/run.js benchmarks mainnetBlocks:10

Benchmarking started...
Running 'mainnetBlocks':
Total number of blocks in data set: 15
Number of blocks to sample: 10
Block 9422905 x 10,661 ops/sec ±3.33% (7[8](https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/3049324777/jobs/4915279042#step:6:9) runs sampled)
Block [9](https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/3049324777/jobs/4915279042#step:6:10)422906 x [10](https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/3049324777/jobs/4915279042#step:6:11),480 ops/sec ±4.78% (78 runs sampled)
Block 9422907 x 10,645 ops/sec ±2.45% (84 runs sampled)
Block 9422908 x 10,918 ops/sec ±2.67% (79 runs sampled)
Block 9422910 x 10,[16](https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/3049324777/jobs/4915279042#step:6:17)4 ops/sec ±7.85% (77 runs sampled)
Benchmark run finished.
```

**Comment by holgerd77** [comment]
@ZLY201 Hey there, thanks a lot for this PR, the benchmark results look really amazing 🎉, generally your whole data structure library makes a really good impression, congrats on that! 👍

**Comment by holgerd77** [comment]
Closed by #2285 

There will be a couple of weeks until a new release on this.
